Competition2018/src/main/java/competition/subsystems/drive/commands/DriveForDistanceCommand.java

```diff
@@ -0,0 +1,110 @@
+package competition.subsystems.drive.commands;
+
+import com.google.inject.Inject;
+
+import xbot.common.command.BaseCommand;
+import xbot.common.logging.RobotAssertionManager;
+import xbot.common.math.ContiguousHeading;
+import xbot.common.math.PIDManager;
+import xbot.common.math.PIDFactory;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.pose.PoseSubsystem;
+
+public class DriveForDistanceCommand extends BaseCommand {
+
+    private final PIDManager travelManager;
+    private final PIDManager headingDrivePid;
+    private final PoseSubsystem poseSubsystem;
+    private final DriveSubsystem drive;
+
+    private double deltaDistance;
+    private double targetDistance;
+    private double startingPosition;
+    public final double defaultPValue = 1 / 80d;
+
+    private DoubleProperty deltaDistanceProp;
+    private ContiguousHeading targetHeading;
+
+    @Inject
+    public DriveForDistanceCommand(DriveSubsystem driveSubsystem, XPropertyManager propManager,
```

Comment:
A lot of the "go straight" code in here can be replaced by the HeadingAssistModule.